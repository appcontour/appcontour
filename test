#!/bin/bash
#

l__ist=`ls tests/test.??`
if [ -n "$1" ]
then
  l__ist=tests/test.$1
fi

tmpf__ile=test$$.tmp
failed="0"

for f__ile in $l__ist
do
  i__nfo=$(grep "^#testname:" $f__ile | cut -f2- -d:)
  if [ -n "$i__nfo" ]; then i__nfo=" ($i__nfo)"; fi
  test=`echo $f__ile | cut -f2 -d\.`
  echo "Running test $test$i__nfo..."
  source $f__ile 2>/dev/null >$tmpf__ile
  exitcode=$?

  if [ -f ${f__ile}.out ]
  then
    if ! diff $tmpf__ile ${f__ile}.out
    then
      failed=$[ $failed + 1 ]
    fi
  else
    echo "...new test, copying out file"
    cp $tmpf__ile ${f__ile}.out
  fi
  rm $tmpf__ile
  echo "$exitcode" >$tmpf__ile
  if [ -f ${f__ile}.ec ]
  then
    if ! diff $tmpf__ile ${f__ile}.ec
    then
      failed=$[ $failed + 1 ]
    fi
  else
    echo "...new test, copying ec file"
    cp $tmpf__ile ${f__ile}.ec
  fi
  rm $tmpf__ile
done

if [ "$failed" -gt "0" ]
then
  echo $failed tests failed
  exit 1
fi

exit 0
