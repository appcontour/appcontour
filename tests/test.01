./contour applyrule n1 <examples/tre_sfere.morse
