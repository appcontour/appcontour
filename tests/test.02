#./contour applyrule n1 <examples/three_spheres.morse | ./contour applyrule n1 | ./contour applyrule n1 | ./contour applyrule n1
./contour applyrule k2 <examples/three_spheres.morse | ./contour applyrule k2 | ./contour applyrule k2 | ./contour applyrule k2
