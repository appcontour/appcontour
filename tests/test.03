./contour applyrule n1 <examples/tre_sfere.morse | ./contour applyrule n1 | ./contour applyrule n1
