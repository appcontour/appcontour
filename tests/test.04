./contour ishuffman -q examples/two_rings.morse
./contour iscontour -q examples/immersed/boysurface.morse
