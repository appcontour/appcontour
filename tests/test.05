./contour applyrule n2 <examples/sfera_dietro_toro.morse
./contour applyrule n3 <examples/sfera_davanti_toro.morse
./contour applyrule n4 <examples/due_anelli.morse | ./contour applyrule n2 | ./contour applyrule n3 | ./contour applyrule n1
./contour applyrule n4 <examples/toro2.morse
./contour applyrule N5 <examples/tre_sfere.morse | ./contour applyrule N5:6 | ./contour applyrule N1 | ./contour applyrule N1 | ./contour applyrule N1
