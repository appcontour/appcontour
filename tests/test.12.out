mendes {
Node 0:- 0;
Node 1:+ 0;
Node 2:+ 0;
Node 3:+ 0;
Node 4:+ 0;
Node 5:+ 1;
Node 6:- 1;
Node 7:- 0;
Node 8:- 0;
Node 9:- 0;
Node 10:- 0;
Node 11:+ 0;
Node 12:- 0;
Node 13:+ 0;
Arc 0: 1 10;
Arc 1: 1 12;
Arc 2: 2 0;
Arc 3: 3 7;
Arc 4: 3 12;
Arc 5: 4 8;
Arc 6: 4 12;
Arc 7: 5 6;
Arc 8: 5 10;
Arc 9: 11 9;
Arc 10: 11 12;
Arc 11: 13 0;
Arc 12: 13 9;
Arc 13: 13 10;
}
mendes_contour {
Node 0: (20,1);
Node 1: (22,2);
Node 2: (3,0) (20,0);
Node 3: (11,2) (13,2) (18,4);
Node 4: (12,2) (14,2) (19,4);
Node 5: (17,2) (17,4) (24,2);
Node 6: (2,1) (17,3) (17,5) (24,3);
Node 7: (7,1) (9,3) (11,3) (13,3) (18,5);
Node 8: (8,1) (10,3) (12,3) (14,3) (19,5);
Node 9: (9,1) (10,1) (15,1) (16,1) (18,1) (19,1);
Node 10: (3,1) (4,1) (5,1) (6,1) (17,1) (20,3) (22,1) (24,1) (25,1);
Node 11: (9,0) (10,0) (11,0) (12,0) (15,0) (16,0) (18,0) (19,0) (23,0);
Node 12: (11,1) (12,1) (13,1) (14,1) (15,3) (16,3) (18,3) (19,3) (21,1) (22,3) (23,1);
Node 13: (2,0) (4,0) (5,0) (6,0) (7,0) (8,0) (9,2) (10,2) (13,0) (14,0) (15,2) (16,2) (17,0) (18,2) (19,2) (20,2) (21,0) (22,0) (24,0) (25,0);
Arc 0: 6;
Arc 1: 7;
Arc 2: 3;
Arc 3: 27 28 34 37;
Arc 4: 20 32;
Arc 5: 26 29 35 36;
Arc 6: 21 33;
Arc 7: 8 9 10 11 30 31 38 39;
Arc 8: 4;
Arc 9: 12 13 14 15 16 17 18 19;
Arc 10: 2;
Arc 11: 5;
Arc 12: 22 23 24 25;
Arc 13: 1;
}
